azurerm/internal/services/datafactory/tests/data_factory_integration_runtime_self_hosted_test.go

```diff
@@ -0,0 +1,277 @@
+  role_definition_name = "Contributor"
+  principal_id         = azurerm_data_factory.target.identity[0].principal_id
+}
+
+resource "azurerm_data_factory" "target" {
+  name                = "acctestdfirsht%d"
+  location            = azurerm_resource_group.target.location
+  resource_group_name = azurerm_resource_group.target.name
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_data_factory_integration_runtime_self_hosted" "target" {
+  name                = "acctestirsht%d"
+  data_factory_name   = azurerm_data_factory.target.name
+  resource_group_name = azurerm_resource_group.target.name
+
+  rbac_authorization {
+    resource_id = azurerm_data_factory_integration_runtime_self_hosted.host.id
+  }
+
+  depends_on = [azurerm_role_assignment.target, azurerm_virtual_machine_extension.test]
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.RandomString, data.RandomString, data.RandomInteger, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+}
+
+func testCheckAzureRMDataFactoryIntegrationRuntimeSelfHostedExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).DataFactory.IntegrationRuntimesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		factoryName := rs.Primary.Attributes["data_factory_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Data Factory Self-hosted Integration Runtime: %s", name)
+		}
+
+		resp, err := client.Get(ctx, resourceGroup, factoryName, name, "")
+		if err != nil {
+			return fmt.Errorf("Bad: Get on IntegrationRuntimesClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Data Factory Self-hosted Integration Runtime %q (Resource Group: %q, Data Factory %q) does not exist", name, factoryName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMDataFactoryIntegrationRuntimeSelfHostedDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).DataFactory.IntegrationRuntimesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_data_factory_integration_self_hosted" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		factoryName := rs.Primary.Attributes["data_factory_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
```

Comment:
Could we use a resource specific parse id function here like can be seen in the analysis package: parse.AnalysisServicesServerID